src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -37,6 +37,12 @@ public SafeSocketHandle(IntPtr preexistingHandle, bool ownsHandle)
 #if DEBUG
         private InnerSafeCloseSocket _innerSocketCopy;
 #endif
+        private bool _sentShutdown;
+
+        internal void TrackShutdown(SocketShutdown how)
+            => _sentShutdown = _sentShutdown ||
```

Comment:
> `_sentShutdown = _sentShutdown ||`

What is this trying to do?

Reply:
This tracks if the user has closed the write end of the socket. I'll rename this to hasShutdownSend.

Reply:
No, I mean specifically the part where it's assigning `_sentShutdown` to itself, and using that assignment as part of a logical or.

Reply:
Ah, in case Shutdown is called multiple times, with different values. It continues to keeps track if the send end was shutdown.

Reply:
I still don't understand.  This is doing `a = a`.  Why do the assignment at all?

Reply:
Oh!  I think I'm getting confused by precedence.